flink-core/src/main/java/org/apache/flink/api/common/state/v2/AggregatingStateDescriptor.java

```diff
@@ -72,6 +72,23 @@ public AggregatingStateDescriptor(
         this.aggregateFunction = checkNotNull(aggregateFunction);
     }
 
+    /**
+     * Creates a new state descriptor with the given name, function, and type.
+     *
+     * <p>If this constructor fails (because it is not possible to describe the type via a class),
```

Comment:
It would be good to detail when a type cannot be described by a class, or point to documentation.

Reply:
This is more like java specific. The most simple one is, you can't describe 'Pojo<A>' via a class since it has type parameter A. There are more complex ones. I don't think Flink should detail this.